arch/sim/src/sim/posix/sim_hostfs.c

```diff
@@ -225,7 +225,24 @@ nuttx_ssize_t host_read(int fd, void *buf, nuttx_size_t count)
 {
   /* Just call the read routine */
 
-  nuttx_ssize_t ret = read(fd, buf, count);
+  nuttx_ssize_t ret;
+
+  if (fd == STDIN_FILENO)
+    {
+      static int ifd = -1;
+
+      if (ifd < 0)
+        {
+          ifd = host_open("/dev/tty", O_RDWR | O_NONBLOCK, 0666);
```

Comment:
does this "/dev/tty" mean you are assuming a particular host platform?

Reply:
I think it could be configurable, user could use a USB/Serial or other device file

Reply:
yes, the code has to reopen input with O_NONBLOCCK mode to support the unblocking reading.

Reply:
> I think it could be configurable, user could use a USB/Serial or other device file

@chirping78 let's create a new patch to make it configurable.